tests/Silverback.Integration.Tests/Messaging/Validation/ValidatorConsumerBehaviorTests.cs

```diff
@@ -0,0 +1,205 @@
+                }
+            };
+
+        [SuppressMessage("ReSharper", "InconsistentNaming", Justification = "TestData")]
+        public static IEnumerable<object[]> HandleAsync_MessageValidationModeLogWarning_WarningIsLogged_TestData =>
+            new[]
+            {
+                new object[]
+                {
+                    new TestValidationMessage
+                        { Id = "1", String10 = "123456789abc", IntRange = 5, NumbersOnly = "123" },
+                    "An invalid message has been processed. | validation errors: The field String10 must be a string with a maximum length of 10."
+                },
+                new object[]
+                {
+                    new TestValidationMessage
+                        { Id = "1", String10 = "123456", IntRange = 30, NumbersOnly = "123" },
+                    "An invalid message has been processed. | validation errors: The field IntRange must be between 5 and 10."
+                },
+                new object[]
+                {
+                    new TestValidationMessage { String10 = "123456", IntRange = 5, NumbersOnly = "123" },
+                    "An invalid message has been processed. | validation errors: The Id field is required."
+                },
+                new object[]
+                {
+                    new TestValidationMessage
+                        { Id = "1", String10 = "123456", IntRange = 5, NumbersOnly = "Test1234" },
+                    "An invalid message has been processed. | validation errors: The field NumbersOnly must match the regular expression '^[0-9]*$'."
+                }
+            };
+
+        [Theory]
+        [MemberData(nameof(HandleAsync_None_WarningIsNotLogged_TestData))]
+        public async Task HandleAsync_None_WarningIsNotLogged(IIntegrationMessage message)
+        {
+            var endpoint = TestConsumerEndpoint.GetDefault();
+            endpoint.MessageValidationMode = MessageValidationMode.None;
+
+            var envelope = new InboundEnvelope(
+                message,
+                null,
+                null,
+                new TestOffset(),
+                endpoint,
+                "source-endpoint");
+
+            IRawInboundEnvelope? result = null;
+            await new ValidatorConsumerBehavior(_inboundLogger).HandleAsync(
+                new ConsumerPipelineContext(
+                    envelope,
+                    Substitute.For<IConsumer>(),
+                    Substitute.For<ISequenceStore>(),
+                    Substitute.For<IServiceProvider>()),
+                context =>
+                {
+                    result = context.Envelope;
+                    return Task.CompletedTask;
+                });
+
+            result.Should().NotBeNull();
+            Action act = () => _loggerSubstitute.Received(LogLevel.Warning, null);
+            act.Should().Throw<InvalidOperationException>();
```

Comment:
Why should it throw?? Is this really what you would except?

(Plus your `act` is an assert in reality, the actual test is happening above, where you call `HandleAsync`.)

Reply:
@BEagle1984 I adjusted the tests and added a new method in the LoggerSubstitute:
https://github.com/BEagle1984/silverback/pull/136/commits/c3a459ab54b0333d05c002be0058642d1064f7fa#diff-1d631d928608d461ad1ca8ae05103165709d096595e6fe98a5acf980c89f0b2f